### DISCUSSION TITLE
Is it possible to make custom local cask for XCode?

### DISCUSSION THREAD

**Discussion by mr-kew** [Homebrew/discussions#3870] [Tap maintenance and Homebrew development]
Hello,

so I would like to have a cask that checks the developer XCode download page for a new version when I run `brew upgrade`. If there is, it should download the file at https://download.developer.apple.com/Developer_Tools/Xcode_14.1/Xcode_14.1.xip and extract it to Applications folder.

I understand it cannot be a public cask because of the developer account login.

If I understand it correctly, the taps should be able to create a custom local formula for any github repo, but I don't know if it is possible to make a custom local cask like that?

**Comment by SMillerDev** [comment]
If you manage to make the download work you can make a cask file for anything, the distribution system is irrelevant for that.

**Reply by mr-kew** [reply]
Hmm I see, I can just create a cask file and put it in the `/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks/`.
Thanks

But... making the download work will probably be an impossible task, I don't see any way how to pass the AppleID login through from terminal.
